src/main/java/io/temporal/internal/statemachines/LocalActivityStateMachine.java

```diff
@@ -0,0 +1,304 @@
+      ExecuteLocalActivityParameters localActivityParameters,
+      Functions.Proc2<Optional<Payloads>, Failure> callback,
+      Functions.Proc1<ExecuteLocalActivityParameters> localActivityRequestSink,
+      Functions.Proc1<CancellableCommand> commandSink,
+      Functions.Proc1<StateMachine> stateMachineSink) {
+    return new LocalActivityStateMachine(
+        replaying,
+        setCurrentTimeCallback,
+        localActivityParameters,
+        callback,
+        localActivityRequestSink,
+        commandSink,
+        stateMachineSink);
+  }
+
+  private LocalActivityStateMachine(
+      Functions.Func<Boolean> replaying,
+      Functions.Func1<Long, Long> setCurrentTimeCallback,
+      ExecuteLocalActivityParameters localActivityParameters,
+      Functions.Proc2<Optional<Payloads>, Failure> callback,
+      Functions.Proc1<ExecuteLocalActivityParameters> localActivityRequestSink,
+      Functions.Proc1<CancellableCommand> commandSink,
+      Functions.Proc1<StateMachine> stateMachineSink) {
+    super(STATE_MACHINE_DEFINITION, commandSink, stateMachineSink);
+    this.replaying = replaying;
+    this.setCurrentTimeCallback = setCurrentTimeCallback;
+    this.localActivityParameters = localActivityParameters;
+    this.localActivityRequestSink = localActivityRequestSink;
+    this.callback = callback;
+    explicitEvent(ExplicitEvent.CHECK_EXECUTION_STATE);
+    explicitEvent(ExplicitEvent.SCHEDULE);
+  }
+
+  enum ExplicitEvent {
+    CHECK_EXECUTION_STATE,
+    SCHEDULE,
+    GET_REQUEST,
+    HANDLE_RESPONSE,
+    NON_REPLAY_WORKFLOW_TASK_STARTED
+  }
+
+  enum State {
+    CREATED,
+    REPLAYING,
+    EXECUTING,
+    REQUEST_PREPARED,
+    REQUEST_SENT,
+    RESULT_NOTIFIED,
+    MARKER_COMMAND_CREATED,
+    MARKER_COMMAND_RECORDED,
+    WAITING_MARKER_EVENT,
+    RESULT_NOTIFIED_REPLAYING
+  }
+
+  public static final StateMachineDefinition<State, ExplicitEvent, LocalActivityStateMachine>
+      STATE_MACHINE_DEFINITION =
+          StateMachineDefinition.<State, ExplicitEvent, LocalActivityStateMachine>newInstance(
+                  "LocalActivity", State.CREATED, State.MARKER_COMMAND_RECORDED)
+              .add(
+                  State.CREATED,
+                  ExplicitEvent.CHECK_EXECUTION_STATE,
+                  new State[] {State.REPLAYING, State.EXECUTING},
+                  LocalActivityStateMachine::getExecutionState)
+              .add(
+                  State.EXECUTING,
+                  ExplicitEvent.SCHEDULE,
+                  State.REQUEST_PREPARED,
+                  LocalActivityStateMachine::sendRequest)
+              .add(State.REQUEST_PREPARED, ExplicitEvent.GET_REQUEST, State.REQUEST_SENT)
+              .add(
+                  State.REQUEST_SENT,
+                  ExplicitEvent.NON_REPLAY_WORKFLOW_TASK_STARTED,
+                  State.REQUEST_SENT)
+              .add(
+                  State.REQUEST_SENT,
+                  ExplicitEvent.HANDLE_RESPONSE,
```

Comment:
and is this for explicitly reporting the local activity result? if so, can we have better naming for explicit event here?

Reply:
renamed to HANDLE_RESULT